### ISSUE TITLE
Crash with MacOS app bundling and PyQt6

### ISSUE THREAD

**Issue by Henxun** [Nuitka/Nuitka#2489]
Before submitting an Issue, please review the
[Issue Guidelines](https://github.com/Nuitka/Nuitka/blob/develop/CONTRIBUTING.md#submitting-an-issue).

- Please check whether the bug was already reported or fixed.

- Please check out if the develop version of Nuitka works better for you.

  Download source, packages [from here](https://nuitka.net/doc/download.html) where you will also
  find instructions how to do it via pip or git.

If you want to post a problem/bug, to help us understand and resolve your issue please check that
you have provided at least the information below, and discard up to here:

- Nuitka version, full Python version, flavor, OS, etc. as output by *this exact* command.

  > python -m nuitka --version
1.8.4
Commercial: None
Python: 3.11.5 (v3.11.5:cce6ba91b3, Aug 24 2023, 10:50:31) [Clang 13.0.0 (clang-1300.0.29.30)]
Flavor: CPython Official
Executable: /Users/admin/Downloads/DemoAppPyInstaller/macos/no-datas/pyqt6/venv/bin/python3
OS: Darwin
Arch: arm64
Version C compiler: /usr/bin/clang (clang).

  If you are not providing the full output. The issue cannot be solved and will be considered
  `invalid`. The command outputs more than you think, and we are adding more all the time. We do not
  want to left guessing or trying things out to reproduce your issue.

- How did you install Nuitka and Python
  brew install python
  pip install nuitka, then install into a virtualenv

- The specific PyPI names and versions

  It should be taken from this output if there specific packages involved, e.g. `numpy`, you are
  expected to shorten this to the relevant ones.

  > python -m pip freeze
Nuitka==1.8.4
ordered-set==4.1.0
PyQt6==6.5.2
PyQt6-Qt6==6.5.2
PyQt6-sip==13.5.2
zstandard==0.21.0

app.py
```
print('1111111')
print('111222')
import sys
print('sys')
try:
    from PyQt6.QtWidgets import QMainWindow, QApplication, QLabel
except Exception as e:
    print(e)
print('22222')
print('333333')
class MainWindow(QMainWindow):

    def __init__(self):
        super().__init__()

        self.setWindowTitle("Hello World")
        l = QLabel("My simple app.")
        l.setMargin(10)
        self.setCentralWidget(l)
        self.show()
        print('inited')

if __name__ == '__main__':
    try:
        app = QApplication(sys.argv)
        w = MainWindow()
        app.exec()
    except Exception as e:
        print(e)     
```
commnd:
```
sudo nuitka3 --standalone --lto=no --enable-plugin=pyqt6 --output-dir=out --macos-create-app-bundle --debugger --run app.py
```
error output:
```
Process 29192 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x8)
    frame #0: 0x00000001869bd204 CoreFoundation`_CFGetNonObjCTypeID + 92
CoreFoundation`:
->  0x1869bd204 <+92>:  ldar   x8, [x8]
    0x1869bd208 <+96>:  ubfx   x8, x8, #8, #10
    0x1869bd20c <+100>: b      0x1869bd2a4               ; <+252>
    0x1869bd210 <+104>: cmp    x9, #0x7
Target 0: (app) stopped.
Process 29192 launched: '/Users/admin/Downloads/DemoAppPyInstaller/macos/no-datas/pyqt6/out/app.app/Contents/MacOS/app' (arm64)
(lldb) bt
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x8)
  * frame #0: 0x00000001869bd204 CoreFoundation`_CFGetNonObjCTypeID + 92
    frame #1: 0x000000018688df20 CoreFoundation`CFBundleCopyBundleURL + 24
    frame #2: 0x00000001061d4a5c QtCore`QLibraryInfoPrivate::path(QLibraryInfo::LibraryPath, QLibraryInfoPrivate::UsageMode) + 440
    frame #3: 0x00000001061d50ac QtCore`QLibraryInfoPrivate::path(QLibraryInfo::LibraryPath, QLibraryInfoPrivate::UsageMode) + 2056
    frame #4: 0x000000010620b470 QtCore`___lldb_unnamed_symbol7648 + 876
    frame #5: 0x000000010620aa6c QtCore`___lldb_unnamed_symbol7646 + 68
    frame #6: 0x000000010620cfd4 QtCore`___lldb_unnamed_symbol7659 + 108
    frame #7: 0x0000000106209c58 QtCore`QLoggingCategory::QLoggingCategory(char const*, QtMsgType) + 52
    frame #8: 0x0000000105154de0 QtCore.so`warmUpLocationServices() + 268
    frame #9: 0x0000000105157724 QtCore.so`_GLOBAL__sub_I_qdarwinpermissionplugin_location.mm + 12
    frame #10: 0x00000001864f77c4 dyld`invocation function for block in dyld4::Loader::findAndRunAllInitializers(dyld4::RuntimeState&) const + 156
    frame #11: 0x000000018653bfe0 dyld`invocation function for block in dyld3::MachOAnalyzer::forEachInitializer(Diagnostics&, dyld3::MachOAnalyzer::VMAddrConverter const&, void (unsigned int) block_pointer, void const*) const + 172
    frame #12: 0x000000018652f524 dyld`invocation function for block in dyld3::MachOFile::forEachSection(void (dyld3::MachOFile::SectionInfo const&, bool, bool&) block_pointer) const + 524
    frame #13: 0x00000001864dc2d8 dyld`dyld3::MachOFile::forEachLoadCommand(Diagnostics&, void (load_command const*, bool&) block_pointer) const + 296
    frame #14: 0x000000018652e548 dyld`dyld3::MachOFile::forEachSection(void (dyld3::MachOFile::SectionInfo const&, bool, bool&) block_pointer) const + 192
    frame #15: 0x000000018653113c dyld`dyld3::MachOFile::forEachInitializerPointerSection(Diagnostics&, void (unsigned int, unsigned int, bool&) block_pointer) const + 160
    frame #16: 0x000000018653bba0 dyld`dyld3::MachOAnalyzer::forEachInitializer(Diagnostics&, dyld3::MachOAnalyzer::VMAddrConverter const&, void (unsigned int) block_pointer, void const*) const + 432
    frame #17: 0x00000001864f769c dyld`dyld4::Loader::findAndRunAllInitializers(dyld4::RuntimeState&) const + 176
    frame #18: 0x00000001864f7950 dyld`dyld4::Loader::runInitializersBottomUp(dyld4::RuntimeState&, dyld3::Array<dyld4::Loader const*>&) const + 216
    frame #19: 0x00000001864f7ae8 dyld`dyld4::Loader::runInitializersBottomUpPlusUpwardLinks(dyld4::RuntimeState&) const + 328
    frame #20: 0x0000000186514aa8 dyld`dyld4::APIs::dlopen_from(char const*, int, void*) + 1212
    frame #21: 0x0000000100086efc app`_EXECUTE_EMBEDDED_MODULE [inlined] callIntoExtensionModule(tstate=0x00000001013c6c48, full_name="PyQt6.QtCore", filename="/Users/admin/Downloads/DemoAppPyInstaller/macos/no-datas/pyqt6/out/app.app/Contents/MacOS/PyQt6/QtCore.so") at MetaPathBasedLoader.c:736:20 [opt]
    frame #22: 0x0000000100086e64 app`_EXECUTE_EMBEDDED_MODULE at MetaPathBasedLoader.c:1009:9 [opt]
    frame #23: 0x0000000100086c64 app`_EXECUTE_EMBEDDED_MODULE(tstate=0x00000001013c6c48, module=0x0000000103a1d5d0, module_name=0x0000000103a23470, name="PyQt6.QtCore") at MetaPathBasedLoader.c:1074:18 [opt]
    frame #24: 0x00000001000a941c app`_nuitka_loader_exec_module [inlined] EXECUTE_EMBEDDED_MODULE(tstate=0x00000001013c6c48, module=0x0000000103a1d5d0) at MetaPathBasedLoader.c:1147:12 [opt]
    frame #25: 0x00000001000a9404 app`_nuitka_loader_exec_module(self=<unavailable>, args=<unavailable>, kwds=<unavailable>) at MetaPathBasedLoader.c:1607:12 [opt]
    frame #26: 0x0000000100fa9a20 Python`cfunction_call + 60
    frame #27: 0x0000000100f45738 Python`_PyObject_MakeTpCall + 128
    frame #28: 0x00000001010688e4 Python`_PyEval_EvalFrameDefault + 52820
    frame #29: 0x000000010106e264 Python`_PyEval_Vector + 156
    frame #30: 0x0000000100f4815c Python`object_vacall + 292
    frame #31: 0x0000000100f47f8c Python`PyObject_CallMethodObjArgs + 92
    frame #32: 0x00000001010a6d10 Python`PyImport_ImportModuleLevelObject + 996
    frame #33: 0x00000001010539bc Python`builtin___import__ + 168
    frame #34: 0x0000000100faa430 Python`cfunction_vectorcall_FASTCALL_KEYWORDS + 80
    frame #35: 0x0000000100f470f0 Python`_PyObject_CallFunctionVa + 152
    frame #36: 0x0000000100f4704c Python`PyObject_CallFunction + 48
    frame #37: 0x00000001010a6634 Python`PyImport_Import + 272
    frame #38: 0x00000001010a86b4 Python`PyImport_ImportModule + 60
    frame #39: 0x0000000100e05728 sip.so`sip_api_export_module + 88
    frame #40: 0x0000000105615188 QtWidgets.so`PyInit_QtWidgets + 276
    frame #41: 0x0000000100086f54 app`_EXECUTE_EMBEDDED_MODULE [inlined] callIntoExtensionModule(tstate=0x00000001013c6c48, full_name="PyQt6.QtWidgets", filename="/Users/admin/Downloads/DemoAppPyInstaller/macos/no-datas/pyqt6/out/app.app/Contents/MacOS/PyQt6/QtWidgets.so") at MetaPathBasedLoader.c:773:24 [opt]
    frame #42: 0x0000000100086e64 app`_EXECUTE_EMBEDDED_MODULE at MetaPathBasedLoader.c:1009:9 [opt]
    frame #43: 0x0000000100086c64 app`_EXECUTE_EMBEDDED_MODULE(tstate=0x00000001013c6c48, module=0x0000000100d8a8e0, module_name=0x0000000100bf5670, name="PyQt6.QtWidgets") at MetaPathBasedLoader.c:1074:18 [opt]
    frame #44: 0x00000001000a941c app`_nuitka_loader_exec_module [inlined] EXECUTE_EMBEDDED_MODULE(tstate=0x00000001013c6c48, module=0x0000000100d8a8e0) at MetaPathBasedLoader.c:1147:12 [opt]
    frame #45: 0x00000001000a9404 app`_nuitka_loader_exec_module(self=<unavailable>, args=<unavailable>, kwds=<unavailable>) at MetaPathBasedLoader.c:1607:12 [opt]
    frame #46: 0x0000000100fa9a20 Python`cfunction_call + 60
    frame #47: 0x0000000100f45738 Python`_PyObject_MakeTpCall + 128
    frame #48: 0x00000001010688e4 Python`_PyEval_EvalFrameDefault + 52820
    frame #49: 0x000000010106e264 Python`_PyEval_Vector + 156
    frame #50: 0x0000000100f4815c Python`object_vacall + 292
    frame #51: 0x0000000100f47f8c Python`PyObject_CallMethodObjArgs + 92
    frame #52: 0x00000001010a6d10 Python`PyImport_ImportModuleLevelObject + 996
    frame #53: 0x00000001010539bc Python`builtin___import__ + 168
    frame #54: 0x0000000100faa430 Python`cfunction_vectorcall_FASTCALL_KEYWORDS + 80
    frame #55: 0x0000000100018bc0 app`CALL_FUNCTION(tstate=0x00000001013c6c48, function_object=0x0000000100b88ae0, positional_args=<unavailable>, named_args=<unavailable>) at calling.h:88:24 [opt]
    frame #56: 0x00000001000227cc app`CALL_FUNCTION_WITH_ARGS5(tstate=0x00000001013c6c48, called=0x0000000100b88ae0, args=<unavailable>) at HelpersCallingGenerated.c:3841:24 [opt]
    frame #57: 0x0000000100021fd4 app`IMPORT_MODULE5(tstate=0x00000001013c6c48, module_name=<unavailable>, globals=<unavailable>, locals=<unavailable>, import_items=<unavailable>, level=<unavailable>) at HelpersImport.c:138:31 [opt]
    frame #58: 0x0000000100003ad4 app`modulecode___main__(tstate=0x00000001013c6c48, module=<unavailable>, loader_entry=<unavailable>) at module.__main__.c:1307:31 [opt]
    frame #59: 0x0000000100086cac app`_EXECUTE_EMBEDDED_MODULE at MetaPathBasedLoader.c:1034:28 [opt]
    frame #60: 0x0000000100086c64 app`_EXECUTE_EMBEDDED_MODULE(tstate=0x00000001013c6c48, module=0x0000000100be25c0, module_name=0x0000000100bffcf0, name="__main__") at MetaPathBasedLoader.c:1074:18 [opt]
    frame #61: 0x0000000100086af8 app`IMPORT_EMBEDDED_MODULE(tstate=0x00000001013c6c48, name="__main__") at MetaPathBasedLoader.c:1128:24 [opt]
    frame #62: 0x00000001000083f0 app`EXECUTE_MAIN_MODULE(module_name=<unavailable>) at MainProgram.c:447:12 [opt]
    frame #63: 0x0000000100007ef8 app`main(argc=<unavailable>, argv=<unavailable>) at MainProgram.c:1635:5 [opt]
    frame #64: 0x00000001864dfe50 dyld`start + 2544
(lldb) quit
```

**Comment by kayhayen** [comment]
With `PyQt6==6.5.2` `PyQt6==6.5.1` and it does it for me too, with `PyQt6==6.5.0` it works

**Comment by Henxun** [comment]
> With `PyQt6==6.5.2` `PyQt6==6.5.1` and it does it for me too, with `PyQt6==6.5.0` it works

**Comment by Henxun** [comment]
PyQt6==6.5.0 do works
i am confused ,why PyQt6==6.5.2 works fine on windows

**Comment by kayhayen** [comment]
The packaging for macOS has a thing called frameworks. From my research, it seems that QtCore from PyQt6 now requires these frameworks to be copied rather than just DLL files. So far this has only been observed with PySide6 and the web engine there only, but it might be a thing for PyQt6.

On Windows and Linux there is no such thing. And honestly, I do not know what frameworks contribute over DLLs. But some API calls will just crash in C code if not inside the framework. 

I am not sure if I really care enough about PyQt6 to do this anytime soon, I might just point out to people that newer PyQt6 won't work on macOS.

**Comment by kayhayen** [comment]
Without commercial funding, this will not happen. PyQt6 on macOS will give an error now.

> FATAL: options-nanny: Using module 'PyQt6' (version 6.5.2) with incomplete support due to untrue condition 'use_pyqt6': PyQt6 on macOS is not supported, use PySide6 instead

It seems that no plugin really loads on PyQt6 without the framework stuff working absolutely perfect, and I do not have found the need to do it for anything else, so this is probably a week of my time, that I don't think anybody will fund, or that I am actually that interested in. People should be using PySide6 instead, the change is easy, and Qt company makes official bindings that they support and support with Nuitka in conjunction with me. So there is no way I am doing this for free, giving that I have not been successful with the other issues that Qt6 with threading has as well.

**Comment by kayhayen** [comment]
Should somebody else desire to work on this, in the config the item to remove is this:

```
      - description: 'PyQt6 on macOS is not supported, use PySide6 instead'
        support_info: 'error'
        when: 'macos and use use_pyqt6'
```

You will get a segfault during the function that tries to find the framework of a loading plugin and unexpectedly fails doing so, causing a crash.

**Comment by redstoneleo** [comment]
Also got this on Win11
>FATAL: options-nanny: Error, failed to evaluate condition 'macos and use use_pyqt6'

**Comment by kayhayen** [comment]
Oops, that `use` should not be there.

**Comment by kayhayen** [comment]
It was about to tell you that it's not supported at all anymore, so no big harm, but I will update it in the next hotfix again.

**Comment by kayhayen** [comment]
Thanks for your report, this is fixed on the factory branch, which is a development version under rapid development. You can try it out by going here: https://nuitka.net/doc/factory.html 

Feedback if this is working is very welcome, just please do not share plans of doing it, but rather confirmations or denials of it working.

**Comment by redstoneleo** [comment]
The issue has gone using the  factory branch,

**Comment by armanhak** [comment]
> eate-app-bundle --debugger --run

this works for me, thanks!

**Comment by kayhayen** [comment]
This is now on the develop and the commercial staging branch and part of the latest pre-release, will be part of the 2.1 release that will eventually follow.
